partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rel/RelOpUnpivot.kt

```diff
@@ -58,7 +58,13 @@ internal sealed class RelOpUnpivot : ExprRelation {
     class Strict(private val expr: ExprValue) : RelOpUnpivot() {
 
         override fun struct(): Datum {
-            val v = expr.eval(env.push(Row()))
+            val v = expr.eval(env.push(Row())).let {
+                if (it.type.code() == PType.VARIANT) {
+                    it.lower()
+                } else {
+                    it
+                }
```

Comment:
Could you please leave a comment for why we need to lower here? Thanks, I'm just guessing it's to ensure we have Datum struct rather than a datum variant|struct